packages/mui-material/src/Tabs/Tabs.js

```diff
@@ -820,44 +848,68 @@ const Tabs = React.forwardRef(function Tabs(inProps, ref) {
 
   const conditionalElements = getConditionalElements();
 
+  const [RootSlot, rootSlotProps] = useSlot('root', {
+    ref,
+    className: clsx(classes.root, className),
+    elementType: TabsRoot,
+    externalForwardedProps: {
+      ...externalForwardedProps,
+      ...other,
+      component,
+    },
+    ownerState,
+  });
+
+  const [ScrollerSlot, scrollerSlotProps] = useSlot('scroller', {
+    ref: tabsRef,
+    className: classes.scroller,
+    elementType: TabsScroller,
+    externalForwardedProps,
+    ownerState,
+    additionalProps: {
+      style: {
+        overflow: scrollerStyle.overflow,
+        [vertical ? `margin${isRtl ? 'Left' : 'Right'}` : 'marginBottom']: visibleScrollbar
+          ? undefined
+          : -scrollerStyle.scrollbarWidth,
+      },
+    },
+  });
+
+  const [FlexContainerSlot, flexContainerSlotProps] = useSlot('flexContainer', {
+    ref: tabListRef,
+    className: classes.flexContainer,
+    elementType: FlexContainer,
+    externalForwardedProps,
+    ownerState,
+    getSlotProps: (handlers) => ({
```

Comment:
Why use `getSlotProps` here and not with other handlers?

Where do the `handlers.onKeyDown?.(event)` come from? If it is from the props, then can we extract it and handle this on the `handleKeyDown` definition?

Reply:
The intention of the change solely due to `slotProps.*` could be a function. That's why `getSlotProps` exists from MUI Base to ensure that `handlers` is resolved to an object.

Reply:
> Because only flexContainer slot has onKeyDown logic in this component.

But there are other event handlers, for example `onChange` on the `scroller` slot 🤔

Reply:
I guess you meant `scrollbar` slot and I think your question is "why scrollbar does not have `getSlotProp` given it receive `onChange`" right?

Let me update it.